crates/astria-sequencer/src/mempool/mod.rs

```diff
@@ -292,34 +292,38 @@ impl Mempool {
     /// All removed transactions are added to the CometBFT removal cache to aid with CometBFT
     /// mempool maintenance.
     #[instrument(skip_all)]
-    pub(crate) async fn run_maintenance<F1, O1, F2, O2>(
+    pub(crate) async fn run_maintenance<F1, O1, F2, O2, F3, O3>(
         &self,
         current_account_nonce_getter: F1,
         current_account_balances_getter: F2,
+        transaction_cost_getter: F3,
```

Comment:
I feel we're maybe at the point where we'd be better to just pass a reference to `S: accounts::StateReadExt` here now rather than having three getters.

The getters can make writing tests simpler, but at this stage we're making the production code quite a bit more cumbersome I think.

Reply:
I'm starting to rewrite the unit tests to reflect this change and I'm afraid that it's creating a too-strong coupling between the mempool and how the internals of the app state work. Are you against keeping the getters as is?

Reply:
I think the mempool is very tightly coupled to the state-reads anyway. There might be some benefit in constructing a handler-based system similar to what axum and warp are doing with their type-erased `Handler`s, but that requires some careful planning.

I think at this stage stronger coupling might be acceptable.

What would speak against it? Spinning off mempool as a free-standing crate?

Reply:
Yeah - I was thinking we could create a test-only `State` struct which implemented `accounts::StateReadExt` and `cnidarium::StateRead` (with most methods being `unimplemented!()` since most aren't used), but this won't fly as we already have a blanket impl for `accounts::StateReadExt`.

I guess we need to use an actual `Storage` instance, populate a state delta with the required data for the test, and use the state delta here.  Might be a bit of boilerplate, but I guess we can abstract that to some mempool test fixture struct which can hold the storage and delta and can have methods to allow the required data to be put in the delta.

Reply:
Finished using an actual storage instance! It wasn't as bad as I thought in terms of testing